### DISCUSSION TITLE
I don't get my REST API working

### DISCUSSION THREAD

**Discussion by JeroenP1974** [Budibase/budibase#12858] [Help] [answered]
Hello everybody,

I'm new to Budibase and I could really need you're help.

I want to create a new REST API in Budibase. 

My tenant url is : noirconsulting.budibase.app
App-name : Fontein Hattem

Table-name : Producten

The tables are internal tables of Budibase.

Thank you all for helping.

**Comment by ConorWebb96** [comment]
Hey @JeroenP1974,

Have you looked at the [public api](https://docs.budibase.com/reference/rowsearch) documentation? This shows all the various api endpoints you can use to create an API using Budibase.

Can you provide some screenshots of how you've setup your rest data source within Budibase?

Thanks in advance for the additional information.

**Reply by JeroenP1974** [reply]
Yes I did but it don't know how to use is. 

Here is a screenshot. I used the values a mentioned in the budibase documentation. 

<img width="1343" alt="Scherm­afbeelding 2024-02-09 om 01 48 27" src="https://github.com/Budibase/budibase/assets/157612513/801a87a4-575c-4281-b786-d318ae3e994c">

**Reply by ConorWebb96** [reply, marked as answer]
Looking at your screenshot you actually haven't followed the [api](https://docs.budibase.com/reference/appgetbyid) documentation as the url to retrieve your application information actually looks like this `https://budibase.app/api/public/v1/applications/appId`. From here you need to replace the appID with your apps id.

<img width="472" alt="Screenshot 2024-02-09 at 07 41 34" src="https://github.com/Budibase/budibase/assets/126772285/589eae7a-4f18-4a2a-af41-cab3a6d7743f">

There isn't any values mentioned like that in the documentation. Those are the response results you get back. The only item that mentioned that needs sent is the appId.

I've tried the above call and its worked perfectly for me.

**Reply by JeroenP1974** [reply]
Okay, thanks for your quick replay. But where can I find my appid? The part of URL like it is mentioned in the budibase doc. results in :

{
  "message": "Invalid app ID provided, please check the x-budibase-app-id header.",
  "status": 400
}

**Reply by ConorWebb96** [reply]
It should be in the url of your app when you enter it using Budibase. I've attached an image below to show this.
<img width="466" alt="Screenshot 2024-02-12 at 07 26 33" src="https://github.com/Budibase/budibase/assets/126772285/592d938e-102f-45cd-a0c1-11aef09f13cc">

You can remove the _dev, from it so that it interacts with your production data instead.

**Reply by JeroenP1974** [reply]
Although I think the documentation in this subject is very poor....I got it working. Thank you.
